### PR TITLE
Add RBIs for all ActiveSupport String extensions

### PR DESCRIPTION
**Pull request by amomchilov**
### Type of Change

- [ ] Add RBI for a new gem
- [x] Modify RBI for an existing gem
- [ ] Other: 

### Changes

* Gem name: ActiveSupport
* Gem source: 
    * https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/string
* Gem API doc: ...
* Tapioca version: `Tapioca v0.11.12`
* Sorbet version: `Sorbet typechecker 0.5.11142`

### CODE REVIEW

File: rbi/annotations/activesupport.rbi [hunk 1]
```diff
@@ -271,6 +271,148 @@ class Symbol
...
+  sig { returns(ActiveSupport::Multibyte::Chars) }
+  def mb_chars; end
+
+  sig { params(separator: String, preserve_case: T::Boolean, locale: T.nilable(Symbol)).returns(String) }
+  def parameterize(separator: "-", preserve_case: false, locale: nil); end
+
+  sig { params(count: Integer, locale: T.nilable(String)).returns(String) }
+  def pluralize(count = nil, locale = :en); end
+
+  sig { params(patterns: T.any(String, Regexp)) }
```
Comment (andyw8) [inline rbi/annotations/activesupport.rbi:359]: Should this always return a string?

https://api.rubyonrails.org/classes/String.html#method-i-remove

File: rbi/annotations/activesupport.rbi [hunk 2]
```diff
@@ -271,6 +271,148 @@ class Symbol
...
+  sig { params(count: Integer, locale: T.nilable(String)).returns(String) }
+  def pluralize(count = nil, locale = :en); end
+
+  sig { params(patterns: T.any(String, Regexp)).returns(String) }
+  def remove(*patterns); end
+
+  sig { params(patterns: T.any(String, Regexp)).returns(String) }
+  def remove!(*patterns); end
+
+  sig { returns(T.untyped) }
```
Comment (andyw8) [inline rbi/annotations/activesupport.rbi:365]: Should this return `T.nilable(T.any(Class, Module))`?

Comment (amomchilov) [reply rbi/annotations/activesupport.rbi:368]: Nope! A constant can hold any kind of value, e.g.

```ruby
ABC = 123
"ABC".safe_constantize # => 123
```

**Review by KaanOzkan — APPROVED** [review]
@amomchilov Just a heads up the `activesupport` annotation is ignored in the tapioca config in core so you can't observe any errors. Sigs look OK to me, other applications can benefit from them and report back if there are issues.

**Pull request merged**